Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -38,41 +36,27 @@ void ChangePulsetime::init() {
 /** Execute the algorithm.
  */
 void ChangePulsetime::exec() {
-  EventWorkspace_const_sptr in_ws = getProperty("InputWorkspace");
+  EventWorkspace_sptr in_ws;
+  SpectrumIndexSet indexSet(0);
+
+  std::tie(in_ws, indexSet) = std::tuple<EventWorkspace_sptr, SpectrumIndexSet>(
```

Comment:
Is that because `std::tie` is s tuple of references? The way it is not is too cumbersome I think, we should try to find a better solution.